tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -143,32 +143,50 @@ def _ValidateTranslation(self):
         print('Invalid translation compliance string', translation_body.data)
         return False
 
+    # TODO(charbull): refactor this while refactoring the class
+    # validate fields
+    entity_type_str = str(self.entity['type'])
+    type_parse = entity_type_str.split('/')
+    namespace = type_parse[0]
+    entity_type = type_parse[1]
+    entity_type = self.universe.GetEntityType(namespace, entity_type)
+    all_fields_dict = entity_type.GetAllFields()
+
     # iterate through each translation device key and determine its form
-    for device_name in translation_body.keys():
+    for field_name in translation_body.keys():
       # check if keys are UDMI compliant
-      if isinstance(translation_body[device_name].data, str):
+      if isinstance(translation_body[field_name].data, str):
         continue
 
-      device_map = translation_body[device_name]
+      #check if the field_name is on the type
+      # TODO(charbull), the key in the dictionary all_fields_dict starts with `/`, needs to be cleaned
+      # pop the field out
+      key_field_name = '/'+field_name.data
+      opt_wrapper_field = all_fields_dict.pop(key_field_name, None)
```

Comment:
This is also error prone as written in the case of types that don't use global fields because it's not handling namespace.  

Recall the rules "" ed is config text: 
"FIELD"  could be LOCAL/FIELD or /FIELD depending on whether that field is defined locally in the namespace of the type
"OTHER/FIELD" is always OTHER/FIELD

so the field from the config should be split on / and the parts compared.  For unqualified fields, you need to check /FIELD and LOCAL/FIELD.  

Bonus points for not doing string manipulation other than splitting the value in the config.

Reply:
can you provide an example please?
Not sure I follow